spec/spec_helper.rb

```diff
@@ -39,6 +39,12 @@
   config.append_after(:each, db: true) do |ex|
     ActiveRecord::Base.connection.rollback_transaction
 
-    raise "Migrations are pending: #{ex.metadata[:location]}" if ActiveRecord::Base.connection.migration_context.needs_migration?
+    migration_context = if Rails::VERSION::MAJOR >= 7
+      ActiveRecord::MigrationContext.new("db/migrate")
```

Comment:
`ActiveRecord::MigrationContext` was made a public API in https://github.com/rails/rails/commit/eb09f5997256061987f979b33b0c32d6b66a36a9 released with Rails 7. Prior to that, it was a non-public API (with a different interface)